src/bootstrap/lib.rs

```diff
@@ -1308,7 +1308,9 @@ impl Build {
     }
 
     fn llvm_link_tools_dynamically(&self, target: TargetSelection) -> bool {
-        target.contains("linux-gnu") || target.contains("apple-darwin")
+        target.contains("linux-gnu")
+            || target.contains("linux-musl")
+            || target.contains("apple-darwin")
```

Comment:
tbh, I don't think this function should exist at all - the only place it's used is in https://github.com/rust-lang/rust/blob/b32223fec10743af93913c41ec60645d741b0cb6/src/bootstrap/native.rs#L422-L433 and I don't see why it should be a separate flag from `llvm_link_shared`. Can you do the following:
- Delete the call to `llvm_link_tools_dynamically`, just using `llvm_link_shared` unconditionally
- Delete this function
- Add `x86_64-msvc-1` and `x86_64-msvc-2` to the PR CI (see https://github.com/rust-lang/rust/blob/4a7e2fbb7bfdb7ddcb65b6fcf23bb5b272e66d42/src/ci/github-actions/ci.yml#L544-L554 and https://github.com/rust-lang/rust/blob/4a7e2fbb7bfdb7ddcb65b6fcf23bb5b272e66d42/src/ci/github-actions/ci.yml#L274-L288) to make sure they pass.

Reply:
Could you clarify the part about the CI?  Do you want me to trigger those jobs on this PR (if so, how?) or add the `x86_64-msvc-1` and `x86_64-msvc-1` targets to the pull request actions?

Reply:
Yes, trigger those jobs on this PR. You can just copy paste lines 544-554 immediately after line 288 (make sure you fix the indentation), then run `./x.py run expand-yaml-anchors`.

Reply:
Done